pennylane/tape/qscript.py

```diff
@@ -834,25 +834,60 @@ def numeric_type(self) -> Union[type, tuple[type, ...]]:
     # Transforms: QuantumScript to QuantumScript
     # ========================================================
 
-    def copy(self, copy_operations: bool = False) -> "QuantumScript":
+    def copy(
+        self, copy_operations: bool = False, update: Optional[Union[dict, bool]] = False
+    ) -> "QuantumScript":
         """Returns a shallow copy of the quantum script.
```

Comment:
```suggestion
        """Returns a copy of the quantum script. If `update` was passed, the updated attributes are 
        modified, otherwise, all attributes match the original tape. The copy is a shallow copy if 
        `copy_operations` and `update` are both `False`.
```